src/compiler/scanner.ts

```diff
@@ -2559,7 +2550,7 @@ export function createScanner(languageVersion: ScriptTarget, skipTrivia: boolean
                 let regExpFlags = RegularExpressionFlags.None;
                 while (true) {
                     const ch = codePointChecked(pos);
-                    if (ch === CharacterCodes.EOF || !isIdentifierPart(ch, languageVersion)) {
+                    if (!isIdentifierPart(ch, languageVersion)) {

```

Comment:
This check is unnecessary since `isIdentifierPart(CharacterCodes.EOF, languageVersion)` must return false (because `CharacterCodes.EOF <= CharacterCodes.maxAsciiCharacter`).